### ISSUE TITLE
Multiple builds lead to exponential growth of target/

### ISSUE THREAD

**Issue by adamish** [eclipse/jkube#544]
### Description

I'm using jKube maven plugin in external "Dockerfile" as such. I've found that multiple builds quickly fill the disk. 

I've traced this fundamental feature of the default plugin configuration

The plugin creates a tar of the contents of <project>/target/, and then places that at target/docker/<image name>/latest/tmp/docker-build.tar. This is then fed to Docker, and left in place... 
A subsequent build then takes the entire contents of target/ (including the tar) file, and puts it in the tar file... 

```
544K	target/ # build 1
1.3M	target/ # build 2
2.8M	target/ # build 3
5.8M	target/ # build 4
12M	target/ # build 5
```

The file structure in target/ also appears to become more deeply nested after each build
```
./target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/tmp/docker-build.tar

./target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/tmp/docker-build.tar

./target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/tmp/docker-build.tar

./target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/tmp/docker-build.tar

./target/docker/my-image-name/latest/build/maven/target/docker/my-image-name/latest/tmp/docker-build.tar

./target/docker/my-image-name/latest/tmp/docker-build.tar
```

I've repeated this with a very simple configuration
```
<plugin>
    <groupId>org.eclipse.jkube</groupId>
    <artifactId>kubernetes-maven-plugin</artifactId>
    <configuration>
        <images>
            <image>
                <name>my-image-name:%l</name>
                <build>
                    <dockerFile>${project.basedir}/Dockerfile</dockerFile>
                </build>
            </image>
        </images>
    </configuration>
</plugin>
```

### Info
* Eclipse JKube version : 1.0.2
* Maven version 3.6.3

**Comment by manusa** [comment]
Hi @adamish
Thank you for reporting.
Yes, we're aware of this issue, it has to do with the `contextDir` and assembly configuration are inferred when using only the Dockerfile config.

You can currently work around this issue by configuring your project in any of the ways described in this example [docker-file-provided](https://github.com/eclipse/jkube/tree/5703f015643c70a6043a86d4ed6eb27521d5a381/quickstarts/maven/docker-file-provided).

**Comment by manusa** [comment]
Relates to #529

**Comment by manusa** [comment]
This issue should have been addressed in the latest release ([1.2.0](https://github.com/eclipse/jkube/releases/tag/v1.2.0)).

Could you please confirm if you are still facing this?
